activiti-core/activiti-spring-process-extensions/src/main/java/org/activiti/spring/process/model/Extension.java

```diff
@@ -82,18 +83,22 @@ public VariableDefinition getPropertyByName(String name) {
         return null;
     }
 
-    public boolean hasEmptyInputsMapping(String elementId) {
-        ProcessVariablesMapping processVariablesMapping = mappings.get(elementId);
-        return processVariablesMapping != null && processVariablesMapping.getInputs().size() == 0;
+    public boolean hasMapping(String taskId) {
+        return mappings.get(taskId) != null;
     }
 
-    public boolean hasEmptyOutputsMapping(String elementId) {
+    public boolean hasMappingTypeInputs (String elementId) {
         ProcessVariablesMapping processVariablesMapping = mappings.get(elementId);
-        return processVariablesMapping != null && processVariablesMapping.getOutputs().size() == 0;
+        return processVariablesMapping.getMappingType() != null &&
+            (processVariablesMapping.getMappingType().equals(MappingType.MAP_ALL_INPUTS) ||
+            processVariablesMapping.getMappingType().equals(MappingType.MAP_ALL));
     }
 
-    public boolean hasMapping(String taskId) {
-        return mappings.get(taskId) != null;
+    public boolean hasMappingTypeOutputs(String elementId) {
```

Comment:
I'd rename it to `shouldMapAllOutputs`.

Reply:
Again, disagree. For a boolean function, `has`/`is` prefix is a better fit

Reply:
Same as [above](https://github.com/Activiti/Activiti/pull/3392#discussion_r476305360).